### PR TITLE
merge Branch-MakeStats-TestSuite Hong Liang with master

### LINKED ISSUE #223: Missing testsuite for makeStats feature

### PR DESCRIPTION
**Pull request by tanhl2000**
closes #223

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/MakeStatsCommand.java [hunk 1]
```diff
@@ -26,6 +26,7 @@ public class MakeStatsCommand extends Command {
 
     public static final String SHOWING_STATS_MESSAGE = "Opened statistics window.";
     public static final String NO_STATS_MESSAGE = "No person is tagged to event, no statistics to generate.";
+    public static final String INDEX_OUT_OF_BOUNDS_MESSAGE = "Event index given is out of event list range.";
```
Comment (Qiaoran-M) [inline src/main/java/seedu/address/logic/commands/MakeStatsCommand.java]: This message is in the `Messages` class already and you can directly use it from there

File: src/main/java/seedu/address/logic/commands/MakeStatsCommand.java [hunk 2]
```diff
@@ -43,6 +44,9 @@ public MakeStatsCommand(Index index, boolean isGenderStatistic) {
     @Override
     public CommandResult execute(Model model) throws CommandException {
         ObservableList<Event> eventList = model.getFilteredEventList();
+        if (eventList.size() <= index.getZeroBased() || index.getZeroBased() < 0) {
+            throw new CommandException(INDEX_OUT_OF_BOUNDS_MESSAGE);
```
Comment (Qiaoran-M) [inline src/main/java/seedu/address/logic/commands/MakeStatsCommand.java]:
```suggestion
            throw new CommandException(Messages.MESSAGE_INVALID_EVENT_DISPLAYED_INDEX);
```

File: src/main/java/seedu/address/logic/commands/MakeStatsCommand.java [hunk 3]
```diff
@@ -59,6 +63,14 @@ public CommandResult execute(Model model) throws CommandException {
...
     }
 
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof MakeStatsCommand // instanceof handles nulls
+                && index.equals(((MakeStatsCommand) other).index) // state check
+                && isGenderStatistic == ((MakeStatsCommand) other).isGenderStatistic);
+    }
+
```
Comment (Qiaoran-M) [inline src/main/java/seedu/address/logic/commands/MakeStatsCommand.java]: I think it's better to move the `equals` method to the end of this class (below all other methods) for consistency with other command classes

File: src/test/java/seedu/address/model/statistics/StatisticsDataListTest.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    @Test
+    public void contains() {
+        StatisticDataList modelDataList = new StatisticDataList();
+
+        //StatisticDataList does not contain the specified StatisticData
+        assertFalse(modelDataList.contains(new StatisticData(TEST_NAME, TEST_VALUE)));
+
+        modelDataList.addStatistic(new StatisticData(TEST_NAME, 1.0));
```
Comment (Qiaoran-M) [inline src/test/java/seedu/address/model/statistics/StatisticsDataListTest.java]:
```suggestion
        modelDataList.addStatistic(new StatisticData(TEST_NAME, TEST_VALUE));
```

File: src/test/java/seedu/address/model/statistics/StatisticsDataListTest.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+        ////StatisticDataList contains the specified StatisticData
```
Comment (Qiaoran-M) [inline src/test/java/seedu/address/model/statistics/StatisticsDataListTest.java]:
```suggestion
        //StatisticDataList contains the specified StatisticData
```

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 6]
```diff
@@ -66,6 +71,11 @@ public class CommandTestUtil {
     public static final String INVALID_EMAIL_DESC = " " + PREFIX_EMAIL + "bob!yahoo"; // missing '@' symbol
     public static final String INVALID_ADDRESS_DESC = " " + PREFIX_ADDRESS; // empty string not allowed for addresses
     public static final String INVALID_GENDER_DESC = " " + PREFIX_GENDER + "unknown";
+    public static final String INVALID_TYPE_DESC = " " + PREFIX_TYPE + "c";
+
+    public static final String STATS_DESC_GENDER = " " + PREFIX_TYPE + VALID_GENDER_TYPE;
```
Comment (Qiaoran-M) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java]:
```suggestion
    public static final String TYPE_DESC_GENDER = " " + PREFIX_TYPE + VALID_GENDER_TYPE;
```

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 7]
```diff
@@ -66,6 +71,11 @@ public class CommandTestUtil {
...
+
+    public static final String STATS_DESC_AGE = " " + PREFIX_TYPE + VALID_AGE_TYPE;
```
Comment (Qiaoran-M) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java]:
```suggestion
    public static final String TYPE_DESC_AGE = " " + PREFIX_TYPE + VALID_AGE_TYPE;
```

File: src/test/java/seedu/address/logic/commands/MakeStatsCommandTest.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+                false, true, false), expectedModel);
+    }
+
+    @Test
+    public void execute_failure() {
+        Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+        //Index out of bounds
+        assertCommandFailure(new MakeStatsCommand(INDEX_TOO_LARGE_EVENT, Boolean.FALSE),
+                model, MakeStatsCommand.INDEX_OUT_OF_BOUNDS_MESSAGE);
```
Comment (Qiaoran-M) [inline src/test/java/seedu/address/logic/commands/MakeStatsCommandTest.java]: Same as above
```suggestion
                model, Messages.MESSAGE_INVALID_EVENT_DISPLAYED_INDEX);
```

**Review by Qiaoran-M — CHANGES_REQUESTED** [review]
Very nice test suite implementation! only a few minor changes needed

File: src/main/java/seedu/address/logic/commands/MakeStatsCommand.java (hunk 1 above)
Comment (tanhl2000) [reply src/main/java/seedu/address/logic/commands/MakeStatsCommand.java]: ok will change!

File: src/main/java/seedu/address/logic/commands/MakeStatsCommand.java (hunk 3 above)
Comment (tanhl2000) [reply src/main/java/seedu/address/logic/commands/MakeStatsCommand.java]: ah ok will change!

File: src/test/java/seedu/address/model/statistics/StatisticsDataListTest.java (hunk 4 above)
Comment (tanhl2000) [reply src/test/java/seedu/address/model/statistics/StatisticsDataListTest.java]: thx for catching!

File: src/test/java/seedu/address/model/statistics/StatisticsDataListTest.java (hunk 5 above)
Comment (tanhl2000) [reply src/test/java/seedu/address/model/statistics/StatisticsDataListTest.java]: will change

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java (hunk 6 above)
Comment (tanhl2000) [reply src/test/java/seedu/address/logic/commands/CommandTestUtil.java]: will change

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java (hunk 7 above)
Comment (tanhl2000) [reply src/test/java/seedu/address/logic/commands/CommandTestUtil.java]: will change!

**Review by Qiaoran-M — APPROVED** [review]
LGTM!

**Pull request merged**